pkg/api/norman/store/ingress/store.go

```diff
@@ -9,44 +10,126 @@ import (
+		return p.Store.Create(apiContext, schema, data)
+	}
+	// TODO: remove when support for k8s 1.18 is dropped
+	return p.legacyIngressProxyStore().Create(apiContext, schema, data)
 }
 
 func (p *Store) Update(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, id string) (map[string]interface{}, error) {
 	formatData(id, data, false)
-	data, err := p.Store.Update(apiContext, schema, data, id)
-	return data, err
+	clusterContext, err := p.downstreamClusterContext(apiContext)
+	if err != nil {
+		return nil, err
+	}
+	if ingresswrapper.ServerSupportsIngressV1(clusterContext.K8sClient) {
+		return p.Store.Update(apiContext, schema, data, id)
+	}
+	// TODO: remove when support for k8s 1.18 is dropped
+	return p.legacyIngressProxyStore().Update(apiContext, schema, data, id)
+}
+
+func (p *Store) List(apiContext *types.APIContext, schema *types.Schema, opt *types.QueryOptions) ([]map[string]interface{}, error) {
+	clusterContext, err := p.downstreamClusterContext(apiContext)
+	if err != nil {
+		return nil, err
+	}
+	if ingresswrapper.ServerSupportsIngressV1(clusterContext.K8sClient) {
+		return p.Store.List(apiContext, schema, opt)
+	}
+	// TODO: remove when support for k8s 1.18 is dropped
+	return p.legacyIngressProxyStore().List(apiContext, schema, opt)
+}
+
+func (p *Store) Watch(apiContext *types.APIContext, schema *types.Schema, opt *types.QueryOptions) (chan map[string]interface{}, error) {
+	clusterContext, err := p.downstreamClusterContext(apiContext)
+	if err != nil {
+		return nil, err
+	}
+	if ingresswrapper.ServerSupportsIngressV1(clusterContext.K8sClient) {
+		return p.Store.Watch(apiContext, schema, opt)
+	}
+	// TODO: remove when support for k8s 1.18 is dropped
+	return p.legacyIngressProxyStore().Watch(apiContext, schema, opt)
+}
+
+func (p *Store) ByID(apiContext *types.APIContext, schema *types.Schema, id string) (map[string]interface{}, error) {
+	clusterContext, err := p.downstreamClusterContext(apiContext)
+	if err != nil {
+		return nil, err
+	}
+	if ingresswrapper.ServerSupportsIngressV1(clusterContext.K8sClient) {
+		return p.Store.ByID(apiContext, schema, id)
+	}
+	// TODO: remove when support for k8s 1.18 is dropped
+	return p.legacyIngressProxyStore().ByID(apiContext, schema, id)
+}
+
+// downstreamClusterContext is a helper for fetching the cluster context from the request.
+// TODO: remove when support for k8s 1.18 is dropped
+func (p *Store) downstreamClusterContext(apiContext *types.APIContext) (*config.UserContext, error) {
+	clusterName := p.ClusterManager.ClusterName(apiContext)
+	return p.ClusterManager.UserContext(clusterName)
+}
+
+// legacyIngressProxyStore returns a norman proxy store object for the
+// extensions/v1beta1/Ingress API that can be used to sidestep the registered
+// networking.k8s.io/v1/Ingress API if the server doesn't support it.
+// TODO: remove when support for k8s 1.18 is dropped
+func (p *Store) legacyIngressProxyStore() types.Store {
+	return proxy.NewProxyStore(
```

Comment:
Is there a reason why we need to create a new proxy store on every call vs having it as a field on the Store?